app/dao/template_categories_dao.py

```diff
@@ -0,0 +1,30 @@
+import uuid
+
+from sqlalchemy import asc
+
+from app import db
+from app.dao.dao_utils import transactional
+from app.models import TemplateCategory
+
+
+@transactional
+def dao_create_template_category(template_category: TemplateCategory):
+    template_category.id = uuid.uuid4()
+    db.session.add(template_category)
+
+
+@transactional
+def dao_update_template_category(template_category: TemplateCategory):
+    db.session.add(template_category)
+
+
+def dao_get_template_category_by_id(template_category_id):
+    return TemplateCategory.query.filter_by(id=template_category_id).one()
+
+
+def dao_get_template_category_by_template_id(template_id):
+    return TemplateCategory.query.join(TemplateCategory.templates).filter_by(id=template_id).one()
```

Comment:
I think you should do this the other way around.. cause the join on template <> template categories will be more exp then getting the Template filtered by id. Also you can add TemplateCategory (not just id) to the Template model, so Template.TemplateCategory should give you the TemplateCategory for that. (use sqlalchemy Relationship)